test/evergreen/code_complexity_analysis.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+import ast
+from collections import defaultdict
+import csv
+from heapq import nlargest
+import json
+import os.path
+import pandas as pd
+
+def get_atlas_compatible_code_statistics(summaryFile, dataFile, outfile='atlas_code_complexity.json'):
+    ''' Generate the Atlas compatible format report. '''
+    atlasFormat = {
+                'Test Name': "Code Complexity",
+                'config': {},
+                'metrics': get_code_complexity(summaryFile, dataFile)
+            }
+
+    dirName = os.path.dirname(outfile)
+    if dirName:
+        os.makedirs(dirName, exist_ok=True)
+
+    with open(outfile, 'w') as outfile:
+        json.dump(atlasFormat, outfile, indent=4)
+
+def get_code_complexity(summaryFile, dataFile):
+    ''' Generate a list of intended contents from the complexity analysis. '''
+    resultList = []
+    complexityDict = {}
+    complexityDict['Average'] = get_average(summaryFile)
+    complexityDict['Stats Ranges'] = get_complexity_ranges_list(dataFile)
+    complexityDict['Top ' + str(topNRegions) + ' Regions'] = get_region_list(dataFile)
+    
+    resultList.append(complexityDict)
+    return (resultList)
+
+def get_region_list(dataFile):
+    ''' Retrieve topNRegions/functions with the highest cyclomatic complexity values. '''
+    top5 = dataFrame.nlargest(topNRegions, 'std.code.complexity:cyclomatic')
+    atlasFormat = {}
+    for index, row in top5.iterrows():
+        atlasFormat[row["region"]] = row["std.code.complexity:cyclomatic"]
+
+    return (atlasFormat)
+
+def get_complexity_ranges_list(dataFile):
+    ''' Retrieve 3 set of cyclomatic complexity ranges, above 30, above 50 and above 90. '''
+    rangesList = [20, 50, 90]
+    atlasFormat = {}
+    for i in rangesList:
+        aboveRangeString = 'Above ' + str(i)
+        atlasFormat[aboveRangeString] = get_complexity_max_limit(dataFile, i)
+
+    return (atlasFormat)
+
+def get_complexity_max_limit(complexity_data_path: str, max_limit:int):
```

Comment:
complexity_data_path --> complexity_data_file

Reply:
Adding a space after `max_limit:` for consistency.

Reply:
Fixed in 3154760ada6586d59d072c5df2e956a40df9eba9